controllers/openshift_controller.go

```diff
@@ -595,212 +388,162 @@ func (r *KataConfigOpenShiftReconciler) processKataConfigDeleteRequest() (ctrl.R
+		mc, err := r.newMCForCR(machinePool)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+		var isMcDeleted bool
+
+		err = r.Client.Get(context.TODO(), types.NamespacedName{Name: mc.Name}, mc)
+		if err != nil && k8serrors.IsNotFound(err) {
+			isMcDeleted = true
 		} else if err != nil {
 			return ctrl.Result{}, err
 		}
 
-		if r.kataConfig.Status.UnInstallationStatus.Completed.CompletedNodesCount != r.kataConfig.Status.TotalNodesCount {
-			r.Log.Info("KataConfig uninstallation: ", "Number of nodes completed uninstallation ",
-				r.kataConfig.Status.UnInstallationStatus.Completed.CompletedNodesCount,
-				"Total number of kata installed nodes ", r.kataConfig.Status.TotalNodesCount)
-			// TODO - we don't need this nil check if we know that pool is always initialized
-			if r.kataConfig.Spec.KataConfigPoolSelector != nil &&
-				r.kataConfig.Spec.KataConfigPoolSelector.MatchLabels != nil && len(r.kataConfig.Spec.KataConfigPoolSelector.MatchLabels) > 0 {
-				if r.clientset == nil {
-					r.clientset, err = getClientSet()
-					if err != nil {
-						return ctrl.Result{}, err
-					}
-				}
-
-				for _, nodeName := range r.kataConfig.Status.UnInstallationStatus.InProgress.BinariesUnInstalledNodesList {
-					if contains(r.kataConfig.Status.UnInstallationStatus.Completed.CompletedNodesList, nodeName) {
-						continue
-					}
-
-					if _, ok := r.kataConfig.Spec.KataConfigPoolSelector.MatchLabels["node-role.kubernetes.io/"+machinePool]; !ok {
-						r.Log.Info("Removing the kata pool selector label from the node", "node name ", nodeName)
-						node, err := r.clientset.CoreV1().Nodes().Get(context.Background(), nodeName, metav1.GetOptions{})
-						if err != nil {
-							return ctrl.Result{}, err
-						}
-
-						nodeLabels := node.GetLabels()
-
-						for k := range r.kataConfig.Spec.KataConfigPoolSelector.MatchLabels {
-							delete(nodeLabels, k)
-						}
-
-						node.SetLabels(nodeLabels)
-						_, err = r.clientset.CoreV1().Nodes().Update(context.Background(), node, metav1.UpdateOptions{})
-
-						if err != nil {
-							return ctrl.Result{}, err
-						}
-					}
-				}
+		if !isMcDeleted {
+			err = r.Client.Delete(context.TODO(), mc)
+			if err != nil {
+				// error during removing mc, don't block the uninstall. Just log the error and move on.
+				r.Log.Info("Error found deleting machine config. If the machine config exists after installation it can be safely deleted manually.",
+					"mc", mc.Name, "error", err)
 			}
+			// Sleep for MCP to reflect the changes
+			r.Log.Info("Pausing for a minute to make sure worker mcp has started syncing up")
+			time.Sleep(60 * time.Second)
 		}
 
-		r.Log.Info("Making sure parent MCP is synced properly, KataNodeRole=" + machinePool)
-		if _, ok := r.kataConfig.Spec.KataConfigPoolSelector.MatchLabels["node-role.kubernetes.io/"+machinePool]; ok {
-			mc, err := r.newMCForCR(machinePool)
-			var isMcDeleted bool
+		workreMcp := &mcfgv1.MachineConfigPool{}
```

Comment:
curious why it's `workreMcp` and not `workerMcp`?

Reply:
I think it's a typo we have never fixed. The patch is already huge and I tried to avoid fixing all the stuff I find along the way :-)